config/read.go

```diff
@@ -279,9 +281,18 @@ func CreateHTTPClient(conf ServerConfig) *http.Client {
 		transport.TLSClientConfig = &tls.Config{MinVersion: tls.VersionTLS12}
 	}
 
-	return &http.Client{
-		Timeout:   120 * time.Second,
-		Transport: transport,
+	if retry {
+		client := retryablehttp.NewClient()
+		client.HTTPClient.Timeout = 120 * time.Second
+		client.HTTPClient.Transport = transport
```

Comment:
It may be a good idea to explicitly set the number of retries here, so that we have it explicitly defined and protect against any unexpected changes by upstream. I think the upstream defaults (https://github.com/hashicorp/go-retryablehttp/blob/v0.7.0/client.go#L470) seem sensible, so something like this could work:

```
client.RetryWaitMin = 1 * time.Second
client.RetryWaitMax = 30 * time.Second
client.RetryMax = 4
```

Reply:
Oh, and I think we should also consider setting the Logger variable (and have it passed to this function call, per my comment below)